doc/api/assert.md

```diff
@@ -298,6 +298,65 @@ parameter is undefined, a default error message is assigned. If the `message`
 parameter is an instance of an [`Error`][] then it will be thrown instead of the
 `AssertionError`.
 
+## assert.doesNotReject(block[, error][, message])
+<!-- YAML
+added: REPLACEME
+-->
+* `block` {Function}
+* `error` {RegExp|Function}
+* `message` {any}
+
+> Stability: 1 - Experimental
+
+Awaits for the promise returned by function `block` to complete and not be rejected. See [`assert.rejects()`][] for more details.
+
+When `assert.doesNotReject()` is called, it will immediately call the `block`
+function, and awaits for completion.
+
+If rejected error has the same type as that specified by the `error`
+parameter, then an `AssertionError` is thrown. If the error is of a different
+type, or if the `error` parameter is undefined, the error is propagated back
+to the caller.
+
+The following, for instance, will throw the [`TypeError`][] because there is no
+matching error type in the assertion:
+
+```js
+await assert.doesNotReject(
+  async () => {
+    throw new TypeError('Wrong value');
+  },
+  SyntaxError
+);
+```
+
```

Comment:
Maybe the next lines (copied from `assert.doesNotThrow()` doc) may be skipped in favour of a reference?